XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/SimulatedHandControllerState.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Controllers.Simulation;
+using XRTK.Definitions.Utilities;
+using XRTK.Providers.Controllers.Hands;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Simulation.Hands
+{
+    /// <summary>
+    /// Internal class to define current hand data and smoothly animate hand data points.
+    /// </summary>
+    public class SimulatedHandControllerState
+    {
+        private readonly SimulatedControllerDataProviderProfile profile;
```

Comment:
Why does a controller state need to know about any profile? These should only ever be referenced in the services that's consumed by them. If you need access to this data then query the service directly via properties set by the constructor, or initialized after startup.

Reply:
The controller needs the profile reference, because it contains the configured recorded hand pose s to simulated.

Reply:
That violates the architecture of the system. Let's set the data in properties of the service and utilize it that way. 

Profiles are only primers for the services and shouldn't be used/referenced after the service is initialized.

Reply:
I understand. My intention was to keep "Hand Simulation" specific things away from the Simulation Data Provider, since it is not tied to hands simulation only. I'll refactor this.

Reply:
It goes against the architecture. Profiles are just primer or seed data for the services to use to get setup and running. From there the values can be updated or changed to whatever they need to be.

Let's add the properties that other things need to read from the service and initialize those values using the profiles.